java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiSliceScanner.java

```diff
@@ -172,7 +182,11 @@ public int getNext() throws IOException {
                     if (fieldData == null) {
                         scanData(i, null);
                     } else {
-                        Object fieldValue = ((PrimitiveObjectInspector) fieldInspectors[i]).getPrimitiveJavaObject(fieldData);
+                        Object fieldValue =
+                                ((PrimitiveObjectInspector) fieldInspectors[i]).getPrimitiveJavaObject(fieldData);
+                        if (HudiScannerUtils.isInt64Timestamp(getType(i))) {
+                            fieldValue = HudiScannerUtils.getTimestamp(fieldValue, getType(i));
+                        }
```

Comment:
`((PrimitiveObjectInspector) fieldInspectors[i]).getPrimitiveJavaObject(fieldData);`
Here is the tricky place of the original implementation

For INT64 timestamp, `fieldInspectors[i].getPrimitiveJavaObject(fieldData)` will throw a CastClassException because the real type of `fieldData` (LONG) is not the fieldInspectors expected (TIMESTAMP).

Hive (version < 4.0) will only treat INT64 as a LONG number instead of TIMESTAMP and the cast is occured [here](https://github.com/apache/hive/blob/master/serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/WritableTimestampObjectInspector.java#L38)

So we need handle original `fieldData` before call `fieldInspector.getPrimitiveJavaObject` to bypass this issue for timestamp.

```
// INT64 timestamp type
if (HudiScannerUtils.isInt64Timestamp(getType(i))) {
    fieldValue = HudiScannerUtils.getTimestamp(fieldData, getType(i));
} else {
    fieldValue = ((PrimitiveObjectInspector) fieldInspectors[i]).getPrimitiveJavaObject(fieldData);
}
scanData(i, fieldValue);
```